symfit/core/minimizers.py

```diff
@@ -106,36 +108,29 @@ def __init__(self, *args, **kwargs):
         super(GlobalMinimizer, self).__init__(*args, **kwargs)
 
 
-def ChainedMinimizer(specific_minimizers):
-    class SpecificChainedMinimizer(BaseMinimizer):
-        minimizers = list(specific_minimizers)
-
-        def __init__(self, *args, **kwargs):
-            super(SpecificChainedMinimizer, self).__init__(*args, **kwargs)
-            for idx, minimizer in enumerate(self.minimizers):
-                if not isinstance(minimizer, BaseMinimizer):
-                    # TODO: figure out how to do the initialisation proper and
-                    # pass e.g. the jacobian the minimizers that can deal with 
-                    # it. See also Fit.
-                    self.minimizers[idx] = minimizer(*args, **kwargs)
-
-        def execute(self, minimizer_kwargs=None):
-            if minimizer_kwargs is None:
-                minimizer_kwargs = [{} for _ in self.minimizers]
-            answers = []
-            next_guess = self.initial_guesses
-            for minimizer, kwargs in zip(self.minimizers, minimizer_kwargs):
-                minimizer.initial_guesses = next_guess
-                ans = minimizer.execute(**kwargs)
-                next_guess = ans.params.values()
-                answers.append(ans)
-            final = answers[-1]
-            # TODO: Compile all previous results in one, instead of just the
-            # number of function evaluations. But there's some code down the
-            # line that expects scalars.
-            final.infodict['nfev'] = sum(ans.infodict['nfev'] for ans in answers)
-            return final
-    return SpecificChainedMinimizer
+class ChainedMinimizer(BaseMinimizer):
+    @keywordonly(minimizers=None)
+    def __init__(self, *args, **kwargs):
+        minimizers = kwargs.pop('minimizers')
+        super(ChainedMinimizer, self).__init__(*args, **kwargs)
+        self.minimizers = list(minimizers)
```

Comment:
Why cast to list? If it walks like a duck?

Reply:
Remnant of the previous version